PREP_SYSTEM_FOR_DIETPI.sh

```diff
@@ -537,38 +546,33 @@ _EOF_
 	#	RPi
 	if (( $G_HW_MODEL < 10 )); then
 
-		aPACKAGES_REQUIRED_INSTALL+=('libraspberrypi-bin')
-		aPACKAGES_REQUIRED_INSTALL+=('libraspberrypi0')
-		aPACKAGES_REQUIRED_INSTALL+=('raspberrypi-bootloader')
-		aPACKAGES_REQUIRED_INSTALL+=('raspberrypi-kernel')
-		aPACKAGES_REQUIRED_INSTALL+=('raspberrypi-sys-mods')
-		aPACKAGES_REQUIRED_INSTALL+=('raspi-copies-and-fills')
+		G_AGI libraspberrypi-bin libraspberrypi0 raspberrypi-bootloader raspberrypi-kernel raspberrypi-sys-mods raspi-copies-and-fills
 
 	#	Odroid C2
 	elif (( $G_HW_MODEL == 12 )); then
 
-		aPACKAGES_REQUIRED_INSTALL+=('linux-image-arm64-odroid-c2')
+		G_AGI linux-image-arm64-odroid-c2
 
-	#	Odroid XU3/4
+	#	Odroid XU3/4/HC1
 	elif (( $G_HW_MODEL == 11 )); then
 
-		#aPACKAGES_REQUIRED_INSTALL+=('linux-image-4.9-armhf-odroid-xu3')
-		aPACKAGES_REQUIRED_INSTALL+=('linux-image-armhf-odroid-xu3')
+		G_AGI linux-image-4.9-armhf-odroid-xu3
```

Comment:
4.9 testing kernel, is no longer supported via HK or developed, they have switched to 4.14: http://com.odroid.com/sigong/blog/blog_list.php?bid=192

Meveric will provide packages of 4.14 when stable enough for end users. We can then change as needed in future.

Please revert kernel to 3.x (stable, official) ```linux-image-armhf-odroid-xu3```. Users can install 4.9 testing kernel, if needed manually afterwards.

Reply:
@Fourdee 
> Please revert kernel to 3.x (stable, official)

I think, this was my flaut. Yes I have read the HK kernel story. https://github.com/Fourdee/DietPi/issues/1285#issuecomment-354602923

Hmmm ... but Debian Stretch with Kernel 3.x ?

https://wiki.debian.org/DebianStretch#Packages_.26_versions

> kernel : Linux 4.9

Have I totally missed something?

Reply:
@k-plan @Fourdee 
Yes, still using 3.x also sounds unreasonable to me. If 4.9 is not supported any more, than 3.x won't be either (for Stretch+), even that Meveric still use this as from naming kinda default.

Meveric does already have 4.14 in it's repo, constantly updating (at least there are already several minor versions), so then I would directly go to 4.14 instead of shipping 3.X and overwriting users/base images newer kernel with this.

Reply:
Huh ... as I said before  [Linux Kernel 4.14 Debugging Party](https://forum.odroid.com/viewforum.php?f=146l)

I really don't know ... but it will be your arbitration.

Reply:
nothing to do with 'HW_MODEL == 11'

but on the other hand I can't see any support from HK for newer kernels then 3,x for

Odroid C1/C1+/C0 or C2

can be a argument to stay on 3.x for Odroid XU4(XU3(HC1(MC1 as well.

Reply:
Hmm as for Odroid, the debugging party is over since October, but some people seem to face issues with USB hard drives and UAS enabled. I am not really sure, that UAS is, seems to be a new (faster) USB transfer method or something...

Reply:
UAS => come into focus with Odroid HC1 on board SATA Bridge:
>  JMicron JMS578 USB 3.0 to SATA Bridge with UAS capability to archive near ~300MB/sec transter speed

Reply:
We could also make a separation between the devices, but as at least Odroid claims all to be compatible vice versa...
For now I will revert to 3.x but include a check to keep newer kernel, if installed, as kernel downgrade is also risky/unsupported and this enables us to also test newer kernels by choosing the related base image.

Reply:
Tested on hardware Odroid HC1 (XU4)  with based on `(NR_4.x)_DietPi_v158_OdroidXU4-armv7-(Jessie).img` and it works! 😃 

> For now I will revert to 3.x but ...

Good idea for now, because the device will not start up after `PREP`.
- easy to fix, because `boot_xu4.ini` from DietPi repro has wrong 
`fatload mmc 0:1 0x44000000 exynos5422-odroidxu3.dtb`
Must have `fatload mmc 0:1 0x44000000 exynos5422-odroidxu4.dtb` for kernel 4.x 
But messing around with such a lot of installation, I know this immediately.

Really nice work 👍 
Will test it later with Meveric Jessie (default Kernel 3.x) image

PS: - what not looking "so nice" is autogenerted `fstab`at the moment.

```
root@DietPi:~# cat /etc/fstab
#Samba Client------------------------------------------------------
#/mnt/samba . Please use dietpi-config and the Networking Options: NAS menu to setup this mount

#FTP Client Mount--------------------------------------------------
#/mnt/ftp_client . Please use dietpi-config and the Networking Options: NAS menu to setup this mount

#NFS Client Mount--------------------------------------------------
#/mnt/nfs_client . Please use dietpi-config and the Networking Options: NAS menu to setup this mount

#TMPFS / MISC ------------------------------------------------------
proc            /proc           proc    defaults                                                                                        0 0
tmpfs                   /tmp                    tmpfs   defaults,noatime,nodev,nosuid,mode=1777                         0 0
tmpfs                   /var/log                tmpfs   defaults,size=20m,noatime,nodev,nosuid,mode=1777        0 0
tmpfs                   /DietPi                 tmpfs   defaults,size=10m,noatime,nodev,nosuid,mode=1777        0 0

#Internal Drives---------------------------------------------------
UUID=5886-8548    /boot    auto    defaults,noatime,nofail,discard,x-systemd.automount    0 0
UUID=a4f12232-1670-43f4-8eca-49bfbe4d01ee    /    auto    defaults,noatime,nofail,discard    0 0
```
But it's only cosmetic.

Reply:
Done. This will basically keep the currently installed kernel, as on working devices there should always be one installed.
So if we have a C1/C2 etc. image with preinstalled 3.X kernel, it will stay. If they run a current XUX/HC1 image with 4.14, it will stay as well.

This is maybe not the best solution, as we loose control about which kernel people use, thus which kernel related errors they might face, but it prevents downgrade at least, which might be an error source as well.